### PR TITLE
Replace old chef uploader code with new rake task

### PR DESCRIPTION
**Pull request by rsalevsky**

### CODE REVIEW

**Comment by tboerger** [comment]
Awesome :+1:

File: scripts/barclamp_mgmt_lib.rb [hunk 1]
```diff
@@ -421,40 +421,10 @@ def bc_install_layout_1_app(from_rpm, bc_path)
...
-    upload_cookbooks_from_dir cookbooks, ['ALL'], log
-    upload_data_bags_from_dir databags, log
-    upload_roles_from_dir roles, log
-  end
-
-  puts "Chef components for (#{components.join(", ")}) (format v1) uploaded."
+# sync the chef parts
+def bc_install_layout_1_chef
+  rails_env = ENV["RAILS_ENV"] || "production"
+  system "cd #{CROWBAR_PATH} && RAILS_ENV=#{rails_env} bin/rake chef:upload:all"
```
Comment (MaximilianMeister) [inline scripts/barclamp_mgmt_lib.rb]: why not just

```ruby
system "RAILS_ENV=#{rails_env} #{CROWBAR_PATH}/bin/rake chef:upload:all"
```

**Comment by MaximilianMeister** [comment]
+1

File: scripts/barclamp_mgmt_lib.rb (hunk 1 above)
Comment (vuntz) [inline scripts/barclamp_mgmt_lib.rb]: Please use `su`to not run this as root. We keep having issues with rake being called as root and creating log files that crowbar then cannot write to.

Comment (vuntz) [inline scripts/barclamp_mgmt_lib.rb]: (see a couple of lines below for how it's done for another rake call)

**Comment by rsalevsky** [comment]
@MaximilianMeister @tboerger Can you please review again? I made some changes after your review.

**Comment by MaximilianMeister** [comment]
I am not sure I understand the second commit, as chef-client is running before that code that you have added

File: scripts/install-chef-suse.sh [hunk 2]
```diff
@@ -708,6 +708,17 @@ EOF
 
 $chef_client
 
+if ! [ -e /opt/dell/crowbar_framework/config/client.pem ]; then
+    (export EDITOR=/bin/true
```
Comment (MaximilianMeister) [inline scripts/install-chef-suse.sh]: do we need an `export` here for just this command?

**Comment by tboerger** [comment]
:+1:

**Comment by jdsn** [comment]
:+1:

**Pull request merged**